gateway/ln-gateway/src/lib.rs

```diff
@@ -172,7 +170,10 @@ impl Gateway {
             .await
             .get(&federation_id.to_string())
             .cloned()
-            .ok_or(GatewayError::UnknownFederation)
+            .ok_or(GatewayError::Other(anyhow::anyhow!(
+                "No federation with id {}",
+                federation_id.to_string()
```

Comment:
`.to_string()` isn't necessary here (assuming it is available because the type implements `Display`)

Reply:
Would be a good idea to implement `Display` for `FederationId` as it's not implemented yet